inst/include/mode/r/helpers.hpp

```diff
@@ -117,12 +117,26 @@ std::vector<double> validate_state(cpp11::sexp r_state,
 }
 
 inline
-bool validate_set_initial_state(SEXP r_state, SEXP r_time,
+std::vector<double> validate_time(cpp11::sexp r_time) {
+  if (r_time == R_NilValue) {
+    return std::vector<double>(0);
+  }
+  cpp11::doubles time = cpp11::as_cpp<cpp11::doubles>(r_time);
+  if (time.size() != 1) {
+    cpp11::stop("Expected 'time' to be a scalar value");
+  }
+  std::vector<double> ret(1);
+  std::copy_n(REAL(time.data()), 1, ret.begin());
+  return ret;
+}
+
+inline
+bool validate_set_initial_state(SEXP r_state, SEXP r_pars, SEXP r_time,
                                 SEXP r_set_initial_state) {
   bool set_initial_state = false;
   if (r_set_initial_state == R_NilValue) {
     set_initial_state = r_state == R_NilValue &&
-                        r_time != R_NilValue;
+        (r_time != R_NilValue || r_pars != R_NilValue);
```

Comment:
the boolean in dust is 

```
r_state == R_NilValue && r_pars != R_NilValue && r_step != R_NilValue
```

but here is

```
r_state == R_NilValue && (r_time != R_NilValue || r_pars != R_NilValue)
```

which is slightly different. I think that the dust one is what we want - if time has not been reset we don't want to set the initial state. Also happy to park this for later and potentially change the dust behaviour if we think that's less consistent

Reply:
Ok, makes sense about time resetting being key; but shouldn't we should reset state whenever time is reset, even if new pars not provided?

Reply:
Probably, yes! That's partly why I figured we could come back to this later :)

Reply:
Ok cool, shall we leave this as is?

Reply:
yup, made an issue for later: https://mrc-ide.myjetbrains.com/youtrack/issue/mrc-3139